lib/hls/hls_classes.js

```diff
@@ -120,9 +120,13 @@ shaka.hls.Tag = class {
   /**
    * Create the string key of the tag.
    *
+   * @param {boolean} keepAllAtributtes
    * @return {string}
    */
-  getTagKey() {
+  getTagKey(keepAllAtributtes) {
+    if (keepAllAtributtes) {
+      return this.toString();
+    }
```

Comment:
Would be good to add some unit test for that.

Reply:
Actually the current tests already cover this, the failure is in the playback, but I don't want to add another playback integration test, because it makes the tests take longer to execute.